trunner/tools/gdb.py

```diff
@@ -106,21 +106,34 @@ def __init__(self, interface: str, target: str):
     def run(self):
         try:
             # Use pexpect.spawn to run a process as PTY, so it will flush on a new line
-            self.proc = pexpect.spawn(
-                "openocd",
-                [
-                    "-f",
-                    f"interface/{self.interface}.cfg",
-                    "-f",
-                    f"target/{self.target}.cfg",
-                    "-c",
-                    "reset_config srst_only srst_nogate connect_assert_srst",
-                    "-c",
-                    "init;reset",
-                ],
-                encoding="ascii",
-                logfile=self.logfile,
-            )
+            if self.target == "zynq_7000":
```

Comment:
IMHO we should not make special cases based on target here - use generic options instead, eg. extra_params or per-feature flag, eg.

possible approach with extra_args - to be defined on per-target basis in particular target implementation

```python
    def __init__(self, interface: str, target: str, extra_args: Optional[List[str]] = None):
        self.target = target
        self.interface = interface
        self.extra_args = extra_args
        # (...)

    def run(self):
        args = [
            "-f",
            f"interface/{self.interface}.cfg",
            "-f",
            f"target/{self.target}.cfg",
        ]

        if self.extra_args:
            args.extend(self.extra_args)
       
        # (...)

```